test/e2e/run-all.js

```diff
@@ -217,7 +231,13 @@ async function main() {
     if (testPath !== '') {
       testPath = testPath.replace('\n', ''); // sometimes there's a newline at the end of the testPath
       console.log(`\nExecuting testPath: ${testPath}\n`);
-      await runInShell('node', [...args, testPath]);
+      const testFileName = testPath.split('/').pop();
+      const isTestChangedOrNew = changedOrNewTests?.includes(testFileName);
```

Comment:
This may produce strange results if this test filename matches a subset of another test filename

Instead perhaps we can return an array from `filterE2eChangedFiles` rather than a string, and check here that this matches a filename exactly.

Reply:
thank you for the suggestion! I'm now returning an array from the filtereE2eChangedFiles function :+1: